BUILD.gn

```diff
@@ -515,7 +515,13 @@ action("create_dist_zips") {
   } else {  # !is_win
     script = "//brave/script/create-dist.py"
 
-    inputs = get_target_outputs(":brave_license_dist_resources")
```

Comment:
The changes in this file achieve the following:

Already before this PR, the dist zip on macOS does not contain a `LICENSE` file. See for example https://github.com/brave/brave-browser/releases/download/v1.68.134/brave-v1.68.134-darwin-x64.zip. There are several reasons for this:

1. `create_dist_zips` has `get_target_outputs(":brave_license_dist_resources")` in its `inputs`.
2. `brave_license_dist_resources` produces `$brave_dist_dir/LICENSE`.
3. `create_dist_zips` does `rebase_path("$brave_dist_dir/LICENSE", brave_dist_dir)`.
4. But, on macOS, `create_dist_zips` then sets `--base-dir=$root_out_dir` instead of `--base-dir=$brave_dist_dir`.

The effect is that, on macOS, `create-dist.py` does not find the `LICENSE` file. The special case in `make_zip(...)` silently ignores this on macOS. (See further below.) On Linux, it raises an error.

The changes presented here make it so the `LICENSE` file is not even attempted to be added to the dist zip. This avoids the error when cross-compiling on Linux.

Reply:
So we're missing a license in macOS builds, the cross-compilation revealed this. Maybe we should fix this? Do we have the license in other platform archives?

Reply:
We don't have `LICENSE` in win archives.
An example: https://github.com/brave/brave-browser/releases/download/v1.70.67/brave-v1.70.67-win32-x64.zip

Reply:
We don't seem to have it for iOS. We do have it for Linux: https://github.com/brave/brave-browser/releases/download/v1.67.108/brave-browser-beta-1.67.108-linux-amd64.zip.

Reply:
In any case, I would prefer to leave adding any potentially missing `LICENSE` files to a separate PR. It's beyond the scope of this PR.